qcodes/dataset/sqlite_base.py

```diff
@@ -684,6 +692,25 @@ def perform_db_upgrade_2_to_3(conn: SomeConnection) -> None:
             log.debug(f"Upgrade in transition, run number {run_id}: OK")
 
 
+def get_db_version_and_newest_available_version(path_to_db: str) -> Tuple[int,
```

Comment:
Why returning `newest_version` with a function? I'd suggest to put `newest_version` at the module level, no?

Reply:
This is actually mainly to avoid confusion with `get_user_version` which is quite different from this function.

As discussed offline, that is not so easy/pretty to put the newest version at the module level. Either it's a constant that gets evaluated at the bottom of the module (which is weirdish) or it's a function that returns `len(_UPGRADE_ACTIONS)`, which seems like overkill.

Reply:
agree. I'd just add one argument for the "overkill" option - it makes sense to have access to the newest version value without the need to `get_db_version` at the same time, for example, for reusing that value in tests. I'm gonna add the "overkill" option in #1389 :)

Reply:
I have no objections to that.